torch/csrc/jit/tensorexpr/cuda_codegen.h

```diff
@@ -76,11 +76,15 @@ class TORCH_API CudaCodeGen : public CodeGen {
  public:
   template <typename... Ts>
   CudaCodeGen(Stmt* stmt, Ts... ts) : CodeGen(stmt, std::forward<Ts>(ts)...) {
+    this->set_device(at::Device(at::kCUDA, at::cuda::current_device()));
```

Comment:
Instead of calling `this->set_device` do:
```
CudaCodeGen(Stmt* stmt, Ts... ts)
 : CudaCodeGen(stmt, std::vector<BufferArg>{ts...}, at::Device(at::kCUDA, at::cuda::current_device())
{}
```
(I may be slightly off on the variadic argument syntax there but that's the gist)